lib/pure/bitops.nim

```diff
@@ -76,26 +73,32 @@ when defined(nimHasalignOf):
     ## Returns ``v``, with the bit at position ``bit`` flipped
     v.flipMask(1.T shl bit)
 
-  macro setBits*(v: typed, bits: varargs[typed]): untyped =
+  template setBits*(v: typed): untyped =
+    ## Returns ``v``, with the bits at positions ``bits`` set to 1
+    discard
+
+  template setBits*(v: typed, bit: typed, bits: varargs[typed]): untyped =
     ## Returns ``v``, with the bits at positions ``bits`` set to 1
-    bits.expectKind(nnkBracket)
-    result = newStmtList()
-    for bit in bits:
-      result.add newCall("setBit", v, bit)
```

Comment:
This for loop is easier to understand than your C++ inspired varargs shift recursion with the ``setBits`` taking a single argument.